hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -313,6 +316,11 @@ public void persistToken(Token token) {
 		if (token.hasRemovedUniqueTokens()) {
 			destroyRemoved(token.removedUniqueTokens(), treasury);
 		}
+		if (token.isDeleted() || token.isBelievedToHaveBeenAutoRemoved()) {
```

Comment:
I am not sure about the `token.isBelievedToHaveBeenAutoRemoved()` logic. Can you point out where we use the same logic in the legacy code